controllers/component_image_controller.go

```diff
@@ -458,12 +461,23 @@ func generateRepositoryName(component *appstudioredhatcomv1alpha1.Component) str
 	return component.Namespace + "/" + component.Spec.Application + "/" + component.Name
 }
 
-func (r *ComponentReconciler) generateImageRepository(ctx context.Context, component *appstudioredhatcomv1alpha1.Component, opts *GenerateRepositoryOpts) (*quay.Repository, *quay.RobotAccount, *quay.RobotAccount, error) {
+func (r *ComponentReconciler) generateImageRepository(
+	ctx context.Context,
+	quayClient quay.QuayService,
+	organization string,
```

Comment:
It doesn't look right to have it in the method params and the struct itself.

Reply:
Removed the organization. To avoid duplicate instantiation of the quayClient, the quayClient is passed from Reconcile.